### PR TITLE
Improve OCaml runtimes compared to C

### PR DESCRIPTION
**Pull request by tbrk**
I was interested in your remarks on OCaml vs C performance for numeric code.

Avoiding Bigarray.Genarray gives an order of magnitude performance improvement on my machine (the type annotations and -unsafe seem to have little effect on this example).

Before:

```
time ./bigarray_test
Sum is 0.595922
Create time 3.09944e-06 s
  Init time 1.19804 s
   Sum time 0.446599 s
1.62user 0.03system 0:01.65elapsed 99%CPU (0avgtext+0avgdata 215836maxresident)k
0inputs+0outputs (0major+53400minor)pagefuls 0swaps
```

After:

```
time ./bigarray_test
Sum is 0.595922
Create time 0.214826 s
  Init time 0.730555 s
   Sum time 0.0553079 s
0.94user 0.06system 0:01.01elapsed 99%CPU (0avgtext+0avgdata 304936maxresident)k
0inputs+0outputs (0major+75924minor)pagefuls 0swaps
```

### CODE REVIEW

File: multidim-arrays/bigarray_test.ml [hunk 1]
```diff
@@ -3,17 +3,28 @@
  * In this example, tensors are represented as arrays of arrays of arrays.
  *)
```
Comment (mfnch) [inline multidim-arrays/bigarray_test.ml:4]: Please, move the changes to a separate file and update the comment above to describe your alternative approach. Note that the original comment (tensors are represented as arrays of arrays of arrays) was wrong. I fixed it in the new branch of the repository.

**Review by mfnch — CHANGES_REQUESTED** [review]
Many thanks for your contribution and sorry for the delay in dealing with your pull request. We created a new branch named [contribs_after_publication](https://github.com/fangohr/paper-supplement-ocaml-performance/tree/contribs_after_publication). We would like to merge your contribution into this branch, so that the master branch remains consistent with the original paper. I think we could change the master's `README.md` to clarify that contributions after the publication of the paper are collected in this other branch. It would be great if you could rebase your pull request on top of this new branch.

Your idea of "mixing" arrays and bigarrays is great. I would, however, add your contribution as a separate ml file so that readers can have a look at all approaches and they can compare all of them separately. Rather than modifying `bigarray_test.ml`, you should copy it to a new file, let's say `array_of_bigarrays_test.ml`. Then, change this file as you did in your initial pull request and update the Makefiles to compile this new test. Once all this is done, I'll re-run all the tests and update the numbers as well. Does this sound reasonable?

File: multidim-arrays/bigarray_test.ml (hunk 1 above)
Comment (tbrk) [reply multidim-arrays/bigarray_test.ml:4]: Sorry for the late response. I have a tight deadline coming up. I will make the suggested changes soon.

**Comment by tbrk** [comment]
Sorry for the delay. I have made the changes that you requested on the new branch (contribs_after_publication) and push -f'ed.

**Comment by mfnch** [comment]
Hi Timothy,
Thanks for your contribution. I merged it on the contribs_after_publication branch. I'll re-run all the examples and upload the new results later this week.
Cheers,
Matteo

**Pull request closed without merging**